datastream/dataset.py

```diff
@@ -66,17 +94,36 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
-    def __add__(self, other):
+    def __add__(self, other: Dataset) -> Dataset:
         return Dataset.concat([self, other])
 
-    def map(self, function):
+    def map(self, function: Callable) -> Dataset:
```

Comment:
Would make things easier to either have a typehint for arg and return of `function` or have it clarified in the docstring

Reply:
Sadly. We cannot have a typehint for arg since it can be anything. We can put 'Any' there but I don't think that helps

Reply:
It's `Callabel[[..., Any]]` right? But it still needs to match the element type of `source` right?

Reply:
It's not super obvious what is happening when you look at the documentation for .map
Perhaps a small example to show its functionality instead?

Reply:
Yes, I don't think the extra hinting helps. We need the chain to match types between them but that's called 'variadic' type hinting and it's not supported.

I can add an example